pkg/plugins/runtime/gateway/generator.go

```diff
@@ -138,6 +150,10 @@ func (g Generator) Generate(ctx xds_context.Context, proxy *core_xds.Proxy) (*co
 		for _, host := range hosts {
 			info.Host = host
 
+			// Ensure that generators don't get duplicate routes,
+			// which could happen after redistributing wildcards.
+			info.Host.Routes = merge.UniqueResources(info.Host.Routes)
```

Comment:
Any reason not to put this uniqueness inside the `RedistributeWildcardRoutes` function? Feels like it would make this simpler.

Reply:
Since deduplicating is an independent operation, I'm leaning towards keeping it separate for now.

Reply:
I understand, nevertheless what is the usecase for using `RedistributeWildcardRoutes` with duplicates?
It feels that if ever we call RedistributeWildcard somewhere else we'll need to do the dedupe which was my motivation for putting the dedupe inside `RedistributeWildcardRoutes`.
That doesn't mean that you can't keep the dedup function.

Reply:
Yeh I see your point. There's a lot of refactoring in the future for this stuff, I'm not very happy with the current structure. Let's leave it for now and assume I'll clean up later :)